docs/source/migration/4.0.mdx

```diff
@@ -3,5 +3,79 @@ title: Migrating to Apollo Kotlin 4.0
 description: From 3.x
 ---
 
+> Note: this page is a work in progress.
+
 * Requires Kotlin Gradle Plugin 1.6 (from 1.5)
 * Dropped `enableCompatibilityMetadataVariant`
+
+## Error handling
+
+### Exception surfacing
+
+In v3, errors like network errors, cache misses, and WebSocket protocol errors were surfaced by throwing exceptions in `ApolloCall.execute()` and in Flows (`ApolloCall.toFlow()`, `ApolloCall.watch()`). This was not ideal because it was a difference in how to handle GraphQL errors vs other errors. Moreover, throwing terminates a Flow and consumers would have to handle re-collection.
```

Comment:
```suggestion
In v3, non-GraphQL errors like network errors, cache misses, and WebSocket protocol errors were surfaced by throwing exceptions in `ApolloCall.execute()` and in Flows (`ApolloCall.toFlow()`, `ApolloCall.watch()`). This was not ideal because it was a difference in how to handle GraphQL errors vs other errors. Moreover, throwing terminates a Flow and consumers would have to handle re-collection.
```